crates/nu-cli/src/data/value.rs

```diff
@@ -162,17 +162,18 @@ pub fn compare_values(
 
     use std::cmp::Ordering;
 
-    let result = match (operator, ordering) {
-        (Operator::Equal, Ordering::Equal) => true,
-        (Operator::NotEqual, Ordering::Less) | (Operator::NotEqual, Ordering::Greater) => true,
-        (Operator::LessThan, Ordering::Less) => true,
-        (Operator::GreaterThan, Ordering::Greater) => true,
-        (Operator::GreaterThanOrEqual, Ordering::Greater)
-        | (Operator::GreaterThanOrEqual, Ordering::Equal) => true,
-        (Operator::LessThanOrEqual, Ordering::Less)
-        | (Operator::LessThanOrEqual, Ordering::Equal) => true,
-        _ => false,
-    };
+    let result = matches!(
+        (operator, ordering),
+        (Operator::Equal, Ordering::Equal)
+            | (Operator::GreaterThan, Ordering::Greater)
+            | (Operator::GreaterThanOrEqual, Ordering::Greater)
+            | (Operator::GreaterThanOrEqual, Ordering::Equal)
+            | (Operator::LessThan, Ordering::Less)
+            | (Operator::LessThanOrEqual, Ordering::Less)
+            | (Operator::LessThanOrEqual, Ordering::Equal)
+            | (Operator::NotEqual, Ordering::Greater)
+            | (Operator::NotEqual, Ordering::Less)
+    );
```

Comment:
This may be one others disagree on.  I like `matches!()` better here because the original `match` has 7 return values, 6 `true` and 1 `false`.  I think it is cleaner to instead of saying "return true in these 6 cases," just say "if the pattern matches any of these, return true`; this is what `matches!()` is doing.

This can be removed if anyone disagrees with it.

Reply:
Nope, I think in this case `matches!` is no worse than previously, so happy to use it here 🙂